### ISSUE TITLE
Build failure with Xcode 12.5

### ISSUE THREAD

**Issue by lewis-smith** [helpscout/beacon-ios-sdk#153]
Thank you for taking the time to submit an issue with the details requested below. Our engineering team actively monitors issues submitted to this repo and will generally respond within 2 business days.

**Technical Details:**

- Beacon install method: 
    [ ] Carthage, 
    [x] CocoaPods, 
    [ ] Swift Package Manager,
    [ ] Manual
- SDK version: Beacon (2.2.0)
- iOS OS version: β14.5
- Device make/model: Simulator
- Impacted user count: 100%

**Actual outcome**

    Beacon did not build:

`Failed to build module 'Beacon' from its module interface; the compiler that produced it, 'Apple Swift version 5.3.1 (swiftlang-1200.0.41 clang-1200.0.32.8)', may have used features that aren't supported by this compiler, 'Apple Swift version 5.4 (swiftlang-1205.0.16.12 clang-1205.0.19.6)'`

**Expected outcome**

    Beacon should build

**Steps to reproduce**

    1… Use Xcode beta 12.5 with pods
    2… Build

**Screenshots/Screen Recording**

https://cln.sh/EjHKeW

**Additional Information**

Just aheads up that this is coming. If this is something specific to my setup please let me know and I can try and figure out more details.

**Comment by ScottPetit** [comment]
Hey 👋 @lewis-smith,

Thanks for the report.  Yea it seems the Swift 5.4 compiler is unhappy/crashing with what I'd consider to be pretty basic Swift.  I'm going to try and get a minimal reproducible example and file bugs at swift.org.

Also I'd imagine (because this Swift code in Beacon isn't really pushing the boundaries of the compiler at all) that these just get fixed in subsequent betas without any changes on our end.

**Comment by rromanchuk** [comment]
If we find a workaround can we update this thread? Have to just remove for now dont have time to dig myself right now.  I. know it's beta, but it still bricks devices on beta that need to be built on.  lets try to figure it out and not just wait for the GM. Also, it's the only the lib i have that hit, so either just bad luck, or perhaps something smelly. 😁 

fyi anyone reading, this is not cocoa build specific, building with spm results the same.

**Comment by lewis-smith** [comment]
I was just trying the new xcode beta, and there is now a more helpful error message.

`Conformance of 'Reachability.NetworkStatus' to 'Equatable' has been explicitly marked unavailable here`

**Comment by hs-gv** [comment]
Thanks for sending this new message over, @lewis-smith  .  We'll forward to the team.

**Comment by andreyz** [comment]
@hs-gv would be great if you'd be able to ship a patch soon, as iOS 14.5 release is imminent.

**Comment by bvirlet** [comment]
Would love to see an update on this as well as we need to build our app with Xcode 12.5.

**Comment by lewis-smith** [comment]
This may be helpful? https://forums.swift.org/t/error-the-compiler-that-produced-it-apple-swift-5-2-2-swiftlang-may-have-used-features-that-are-not-supported-by-this-compiler-apple-swift-5-3-swiftlang/39781/5

**Comment by rromanchuk** [comment]
yo helpscout, can you checkout how you're using reachability. I'm yanking this out, but this ticket is going to get louder, highly doubt its just a xcode beta thing.

**Comment by hs-gv** [comment]
Thanks for adding this @rromanchuk .  Certainly an area we've looked at.  We are still expecting the 12.5GM to address this.

**Comment by lewis-smith** [comment]
Do you have something from Apple confirming that the GM will address it?

**Comment by hs-gv** [comment]
hi @lewis-smith , we would be surprised if it is still broken with the GM.  If it is, we are ready to address it.

**Comment by ScottPetit** [comment]
👋 So Xcode 12.5 RC is now available and hasn't resolved the compiler issue.  We'll be out with a new release that resolves it and update this thread when it's available.

**Comment by rromanchuk** [comment]
Looks like it's due to behavior change/feature  in 12.5 https://developer.apple.com/documentation/Xcode-Release-Notes/xcode-12_5-release-notes

> In the above code, it isn’t valid for findBad() to take a Set<Box> because Set requires that its element type conform to Hashable, but the conformance of Box to Hashable isn’t available before macOS 11.
> 
> Note that using an unavailable protocol conformance is a warning, not an error, to avoid potential source compatibility issues. In prior releases, you could write code that made use of unavailable protocol conformances but worked anyway — if the optimizer had eliminated all runtime dispatch through this conformance, or the code in question was entirely unreachable at runtime.
> 
> Protocol conformances can also be marked as completely unavailable or deprecated, by placing an appropriate @available attribute on the extension:
> 
> @available(*, unavailable, message: "Not supported anymore")
> extension Box : Hashable {}
> 
> @available(*, deprecated, message: "Suggest using something else")
> extension Box : Hashable {}

**Comment by rromanchuk** [comment]
and just making sure you know, since i guess it was never mentioned `Reachability.NetworkStatus` ->  `Reachability.Connection`

```swift
public enum Connection: CustomStringConvertible {
        @available(*, deprecated, renamed: "unavailable")
        case none
        case unavailable, wifi, cellular
        public var description: String {
            switch self {
            case .cellular: return "Cellular"
            case .wifi: return "WiFi"
            case .unavailable: return "No Connection"
            case .none: return "unavailable"
            }
        }
    }
```

**Comment by andreyz** [comment]
@ScottPetit @hs-gv what's ETA on the SDK?

**Comment by shaunmcilroy** [comment]
@andreyz Hopping in for Scott -- we don't put hard timelines on releases to prevent any disappointment but we're very close. We'll update this issue when the release goes out. Thanks for your patience cc @lewis-smith (Saw you reaction)

**Comment by ScottPetit** [comment]
@lewis-smith @rromanchuk @andreyz @bvirlet,

Just writing to let y'all know this should be fixed with Version 2.2.1 of the SDK.  Sorry about the issue and hope y'all enjoy all the new Swift goodies.
